okhttp-sse/src/main/java/okhttp3/internal/sse/ServerSentEventReader.java

```diff
@@ -36,96 +40,119 @@ public ServerSentEventReader(BufferedSource source, Callback callback) {
   boolean processNextEvent() throws IOException {
     String id = lastId;
     String type = null;
-    StringBuilder data = null;
+    Buffer data = new Buffer();
 
     while (true) {
-      long crlf = source.indexOfElement(CRLF);
-      if (crlf == -1L) {
+      long lineEnd = source.indexOfElement(CRLF);
+      if (lineEnd == -1L) {
         return false;
       }
 
-      if (crlf == 0L) {
-        skipCrAndOrLf();
+      switch (source.buffer().getByte(0)) {
```

Comment:
worth trying select() ?

Reply:
We can use select but I don't think it has many advantages. For one, we need 5 variants for each key. For example, `data` is actually `data: `, `data:`, `data\n`, `data\r\n`, and `data\r`. The use of the first char is a cheat since everything we care about starts with a different char: <b>d</b>ata, <b>e</b>vent, <b>i</b>d, <b>r</b>etry, <b>`\r`</b>/`\r\n`, or <b>`\n`</b>.

I started with select before moving to the char. I can try it again though.

Reply:
I guess I could match on just `data` and then do the check for the field name divider though!

Reply:
nah, cause select will consume if it matches, and you don’t want to pick up datacos

Reply:
Ah yeah that's right. I had to do a bunch of math on the crlf position (moreso than this PR does) to handle the case where the key was a prefix of an unknown key.